### DISCUSSION TITLE
creating-a-custom-feature-filter-adding-feature-flags-to-an-asp-net-core-app-part-4/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#19] [Announcements]
# creating-a-custom-feature-filter-adding-feature-flags-to-an-asp-net-core-app-part-4/

In this post I show how to create a custom IFeatureFilter that enables a feature based on the current user's claims.

https://andrewlock.net/creating-a-custom-feature-filter-adding-feature-flags-to-an-asp-net-core-app-part-4/

**Comment by juleedev** [comment]
can you add an example of how to unit test feature flags?  especially if you are using featuregates?

**Comment by punitsinghi** [comment]
Thanks for the article. If one wants to consume TimeWindow as part of custom filter basically I want to enable feature based on certain user type along with the time.  I see I can add Microsoft.Timewindow as part of the custom filter but in order to execute that, looks like I need to get the feature definition and loop through it. I was thinking if my custom filter code gets executed and framework takes care of executing TimeWindow filter.

[a comment by an automated account is not shown]

**Comment by hmendezm** [comment]
Hi Andrew,

Thanks for this article. it is very helpful. I have an Azure function and want to use this code/idea but I do not know how to test ClaimsFeatureFilter inside an async Task<IActionResult>. Do you have an example in how read feature flag filters in azure and get inside in a endpoint?

Best 
H. Mendez
